src/map-style.js

```diff
@@ -40,6 +40,16 @@ export class MapStyle extends HTMLElement {
         target.setAttribute(attribute.nodeName, attribute.nodeValue);
       });
     }
+
+    // use observer to monitor the changes in mapStyle textContent
+    this._observer = new MutationObserver(() => {
+      this.styleElement.textContent = this.textContent;
+    });
+    this._observer.observe(this, {
```

Comment:
I wonder if we only need to monitor the `characterData`; there shouldn't be a subtree or child nodes except textContent

Reply:
I tried removing the `childList` and `subtree`, then just `subtree`, and the test below failed both times, so I guess the answer is : no ;-P

Reply:
Yeah it was pretty odd, I played around also, and it appeared these three were required. Atleast I was able to remove `attributes: true, attributeOldValue: true` which we use in map-feature. Perhaps they can also be removed from map-feature.